core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -772,6 +772,474 @@ public void putDelete(byte[] columnFamily, byte[] columnQualifier, ColumnVisibil
+    }
+
+    // methods for changing the column visibility of a Mutation
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @param colVisLen
+     *          column visibility length
+     * @return a TimestampStep object, advancing the method chain
+     */
+    public TimestampStep visibility(byte[] colVis, int colVisLen) {
+      columnVisibility = colVis;
+      columnVisibilityLength = colVisLen;
+      return this;
+    }
+
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @return a TimestampStep object, advancing the method chain
+     */
+    @Override
+    public TimestampStep visibility(byte[] colVis) {
+      return visibility(colVis, colVis.length);
+    }
+
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @return a TimestampStep object, advancing the method chain
+     */
+    @Override
+    public TimestampStep visibility(ByteBuffer colVis) {
+      return visibility(ByteBufferUtil.toBytes(colVis));
+    }
+
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @return a TimestampStep object, advancing the method chain
+     */
+    @Override
+    public TimestampStep visibility(CharSequence colVis) {
+      return visibility(new Text(colVis.toString()));
+    }
+
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @return a TimestampStep object, advancing the method chain
+     */
+    @Override
+    public TimestampStep visibility(ColumnVisibility colVis) {
+      return visibility(colVis.getExpression());
+    }
+
+    /**
+     * Modifies the column visibility of a mutation.
+     *
+     * @param colVis
+     *          column visibility
+     * @return a TimestampStep object, advancing the method chain
+     */
+    @Override
+    public TimestampStep visibility(Text colVis) {
+      return visibility(colVis.toString().getBytes());
+    }
+
+    // methods for changing the timestamp of a Mutation
+    /**
+     * Modifies the timestamp of a mutation.
+     *
+     * @param ts
+     *          timestamp
+     * @return a MutationStep object, advancing the method chain
+     */
+    @Override
+    public MutationStep timestamp(long ts) {
+      hasTs = true;
+      timestamp = ts;
+      return this;
+    }
+
+    /**
+     * Finalizes the method chain by filling the buffer with the gathered Mutation configuration
+     *
+     * @param val
+     *          value
+     * @param delete
+     *          deletion flag
+     */
+    private Mutation set(byte[] val, boolean delete) {
+      if (buffer == null) {
+        throw new IllegalStateException("Can not add to mutation after serializing it");
```

Comment:
I ran the test with code coverage analysis and only two spots were missed in this new inner class. One was here.